File: packages/titan/analyze-failures.py
import re
from collections import Counter

with open('test-results-final.log', 'r') as f:
    content = f.read()

# Error patterns
error_patterns = {
    'Database Config Missing': len(re.findall(r'Connection configuration is required for undefined', content)),
    'Redis Connection': len(re.findall(r'Redis connection error', content)),
    'Timeout': len(re.findall(r'timed out|timeout', content, re.IGNORECASE)),
    'Health Check Failed': len(re.findall(r'health check.*failed|health check timed out', content, re.IGNORECASE)),
    'NOGROUP Error': len(re.findall(r'NOGROUP', content)),
}

# Test files
test_files = re.findall(r'test/[^\s]+\.spec\.ts', content)
test_file_counts = Counter(test_files)

print('Error Pattern Analysis:')
print('=' * 50)
for key, count in sorted(error_patterns.items(), key=lambda x: -x[1]):
    if count > 0:
        print(f'{key}: {count} occurrences')

print('\nMost Mentioned Test Files:')
print('=' * 50)
for file, count in test_file_counts.most_common(10):
    print(f'{file}: {count} mentions')

print(f'\nTotal unique test files: {len(test_file_counts)}')

File: packages/titan/analyze-failures.js
const fs = require('fs');

const content = fs.readFileSync('test-results-final.log', 'utf8');
const lines = content.split('\n');

const errorPatterns = {
  'Database Config Missing': 0,
  'Redis Connection': 0,
  'Timeout': 0,
  'Health Check': 0,
  'NOGROUP': 0,
};

const testFiles = new Set();
const errors = [];

for (let i = 0; i < lines.length; i++) {
  const line = lines[i];
  
  // Track test files
  const testMatch = line.match(/test\/.*\.spec\.ts/);
  if (testMatch) {
    testFiles.add(testMatch[0]);
  }
  
  // Track error patterns
  if (line.includes('Connection configuration is required for undefined')) {
    errorPatterns['Database Config Missing']++;
  }
  if (line.includes('Redis connection error')) {
    errorPatterns['Redis Connection']++;
  }
  if (line.includes('timed out') || line.includes('timeout')) {
    errorPatterns['Timeout']++;
  }
  if (line.includes('health check')) {
    errorPatterns['Health Check']++;
  }
  if (line.includes('NOGROUP')) {
    errorPatterns['NOGROUP']++;
  }
}

console.log('Error Pattern Analysis:');
console.log('======================');
Object.entries(errorPatterns).forEach(([key, count]) => {
  if (count > 0) {
    console.log(`${key}: ${count} occurrences`);
  }
});

console.log('\nTest Files Involved:');
console.log('===================');
testFiles.forEach(file => console.log(file));

console.log(`\nTotal unique test files: ${testFiles.size}`);
